Samples/ResourceManagement/ContainerRegistry/ManageContainerRegistry/Program.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Docker.DotNet;
+using Microsoft.Azure.Management.ContainerRegistry.Fluent;
+using Microsoft.Azure.Management.ContainerRegistry.Fluent.Models;
+using Microsoft.Azure.Management.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using Microsoft.Azure.Management.Samples.Common;
+using System;
+
+namespace ManageContainerRegistry
+{
+    public class Program
+    {
+        private static readonly Region Region = Region.USEast2;
+
+
+        /**
+         * Azure Container Registry sample for managing container registry.
+         *  - Create an Azure Container Registry to be used for holding the Docker images
+         *  - If a local Docker engine cannot be found, create a Linux virtual machine that will host a Docker engine
+         *      to be used for this sample
+         *  - Use Docker Java to create a Docker client that will push/pull an image to/from Azure Container Registry
+         *  - Pull a test image from the public Docker repo (hello-world:latest) to be used as a sample for pushing/pulling
+         *      to/from an Azure Container Registry
+         *  - Create a new Docker container from an image that was pulled from Azure Container Registry
+         */
+        public static void RunSample(IAzure azure, bool testMode)
+        {
+            string rgName = SdkContext.RandomResourceName("rgACR", 15);
+            string acrName = SdkContext.RandomResourceName("acrsample", 20);
+            string saName = SdkContext.RandomResourceName("sa", 20);
+            Region region = Region.USEast2;
+            string dockerImageName = "hello-world";
+            string dockerImageTag = "latest";
+            string dockerContainerName = "sample-hello";
+            string dockerImageRelPath = "samplesdotnet";
+
+            try
+            {
+                //=============================================================
+                // Create an Azure Container Registry to store and manage private Docker container images
+
+                Utilities.Log("Creating an Azure Container Registry");
+
+                IRegistry azureRegistry = azure.ContainerRegistries.Define(acrName)
+                        .WithRegion(region)
+                        .WithNewResourceGroup(rgName)
+                        .WithNewStorageAccount(saName)
+                        .WithRegistryNameAsAdminUser()
+                        .Create();
+
+                Utilities.Print(azureRegistry);
+
+                RegistryListCredentialsResultInner acrCredentials = azureRegistry.ListCredentials();
+
+                // Skip the remaining of the sample when running in test mode (mocked)
+                if (!testMode)
+                {
```

Comment:
We can't... the following code is the core of the sample, how to use the ACR setup at the beginning of the test and push/pull Docker container images from that resource.